### ISSUE TITLE
Support `execution_mode` of "Organization Default" for `tfe_workspace`

### ISSUE THREAD

**Issue by johnnyplaydrums** [hashicorp/terraform-provider-tfe#1118]
#### Use-cases

Allow a `tfe_workspace` to set `execution_mode` to "Organization Default". Currently the only options provided by this provider are `remote`, `local` or `agent`. TF Cloud recently added a new feature to set a default execution mode for all workspaces, and the workspace settings for execution mode has a new "Organization Default" option.

#### Attempted Solutions

Setting `execution_mode` to Organization Default is currently not possible as the provider only supports `remote`, `local` or `agent`.

#### Proposal

In addition to `remote`, `local` or `agent`, add `default`, which will respect the organization default.

To help you with implementation - via the API, I was able to achieve the "Organization Default" setting with the follow request data:
```
{"data":{"attributes":{"setting-overwrites":{"execution-mode":false,"agent-pool":false}}}}
```
So I imagine the provider can use a similar request body when implementing this feature.

<img width="639" alt="Screenshot 2023-10-27 at 12 46 12 PM" src="https://github.com/hashicorp/terraform-provider-tfe/assets/3979377/86c6fe1a-5bcd-4207-bb81-89de63631e1c">

**Comment by barrettclark** [comment]
Thank you for submitting this @johnnyplaydrums

**Comment by tmatilai** [comment]
Maybe this should be a separate resource? Especially you most probably can't specify a non-existing agent pool when creating an organization, and iterative changes and deployments and especially circular dependencies to the agent pool creations don't sound good to me.

**Comment by barrettclark** [comment]
@tmatilai That's what we were thinking as well. We do accept pull requests if you want to take a run at an implementation.

**Comment by tmatilai** [comment]
Assuming that the underlying SDK has the endpoints, I can try at some point. A bit busy at work at the moment, so it has to be a weekend project. 😄

**Comment by tmatilai** [comment]
And now I realize that this issue is actually requesting to amend the `tfe_workspace`, when I was talking about the `tfe_organization` resource. Sorry. But both would be useful of course. 🙂

**Comment by johnnyplaydrums** [comment]
Thanks for helping clarify @tmatilai! You are correct that this request is for a small modification to the `tfe_workspace` resource, to support setting the `default` execution mode.

**Comment by tmatilai** [comment]
> this request is for a small modification to the `tfe_workspace` resource, to support setting the `default` execution mode.

Unfortunately the API is not that simple. Normal one-to-one mapping between Terraform and API resources would suggest adding a `setting_overwrite` block. But IMHO that's not nice to use, so I still suggest a special value and some logic. PR: #1143

If that's approved, I can add similar setup to the `tfe_workspace` data source.

**Comment by johnnyplaydrums** [comment]
Thank you @tmatilai! I was also bummed to see how the API implements this, instead of simply allowing just the `execution-mode` attribute to be updated. At least your PR improves that UX for users of the terraform provider!

**Comment by tmatilai** [comment]
This is finally implemented in #1137.
I.e. the organizational default is used if `execution_mode` is not specified, or set to `null`.

**Comment by barrettclark** [comment]
I'll close this since it's addressed with #1137. Thank you all!

**Comment by johnnyplaydrums** [comment]
Thanks so much for implementing this y'all! Can't wait to test it once a new release is cut - how soon do you think that might be. Thanks again 🙏

**Comment by brandonc** [comment]
@johnnyplaydrums This was released in [v0.51.0](https://github.com/hashicorp/terraform-provider-tfe/releases/tag/v0.51.0)
